### PR TITLE
Fixed #25972 -- Restored support for the isnull lookup with ForeignObject

### PR DESCRIPTION
**Pull request by tomo-otsuka**
Fixes https://code.djangoproject.com/ticket/25972

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
Unless I missed something, the test doesn't seem to be a proper regression test as it passes before the fix is applied, including on 1.8. You may add release notes in `docs/releases/1.9.1.txt` and squash commits when updating, thanks!

**Comment by rogerhu** [comment]
PR'd here: https://github.com/tomo-otsuka/django/pull/3/files

The test requires a foreignobject with multiple to/from fields.

**Comment by tomo-otsuka** [comment]
Updated tests and release notes.

**Comment by timgraham** [comment]
Would some additional assertions for `isnull=False` be useful? 
```
Membership.objects.create(membership_country=self.usa, person=self.bob, group=self.cia)
self.assertQuerysetEqual(
    Membership.objects.filter(person__isnull=False),
    ['<Membership: Bob is a member of CIA>']
)
```
Also the case where `isnull=True` returns some results?

File: docs/releases/1.9.1.txt [hunk 1]
```diff
@@ -9,6 +9,9 @@ Django 1.9.1 fixes several bugs in 1.9.
 Bugfixes
 ========
 
+* Fixed ``isnull`` filtering for ``ForeignObject()``
```
Comment (timgraham) [inline docs/releases/1.9.1.txt]: minor tweak:
```
* Fixed the ``isnull`` query lookup for ``ForeignObject``
```
(also, please put the note at the bottom of the list)

File: docs/releases/1.9.1.txt [hunk 2]
```diff
@@ -64,3 +64,7 @@ Bugfixes
 
 * Restored the functionality of the admin's ``list_editable`` add and change
   buttons (:ticket:`25903`).
+
+* Fixed ``isnull`` query lookup for ``ForeignObject()``
```
Comment (timgraham) [inline docs/releases/1.9.1.txt]: no parentheses for "ForeignObject" and no blank line at the end of the life please

File: tests/foreign_object/tests.py [hunk 3]
```diff
@@ -395,6 +395,15 @@ def test_batch_create_foreign_object(self):
         objs = [Person(name="abcd_%s" % i, person_country=self.usa) for i in range(0, 5)]
         Person.objects.bulk_create(objs, 10)
 
+    def test_isnull_foreign_object(self):
+        self.assertEqual(list(Membership.objects.filter(person__isnull=True)), [])
+
+        Membership.objects.create(membership_country=self.usa, person=self.bob, group=self.cia)
+        self.assertQuerysetEqual(
+                Membership.objects.filter(person__isnull=False),
```
Comment (timgraham) [inline tests/foreign_object/tests.py]: deindent these 3 lines

**Comment by timgraham** [comment]
Can we add a test for the case where isnull=True returns some results?

**Comment by timgraham** [comment]
merged in 8b6974a6857bdc48ad50bf21527b840c27648891, thanks!

**Pull request closed without merging**